### ISSUE TITLE
Step deleted in patched tensorboard log

### ISSUE THREAD

**Issue by Danmou** [wandb/client#613]
When a summary is added through the patched tensorboard summary writer, the step is always removed before calling `wandb.log`. This results in the following warning:
```
wandb: WARNING wandb.log called without a step keyword argument.  Pass the same step that tensorboard is using to avoid data loss see:
wandb: WARNING https://docs.wandb.com/docs/integrations/tensorboard.html#custom-metrics
```
The warning only gets printed once, so I originally thought the warning was only due to logging the graph without a step, see #571, but in fact the step passed to `tf_writer.add_summary()` is always ignored.

I tracked down the problem to the following lines:
https://github.com/wandb/client/blob/b90c8a8e4ee7d10d0f0bafdcac2221d70236b8fc/wandb/tensorboard/__init__.py#L162-L163

Why is it like this? I don't really mind that wandb does its own step counting, but it's a bit confusing and the warning is downright misleading if this is the intended behaviour.

**Comment by vanpelt** [comment]
Hey @danmou, I have a branch that removes this warning we’ll be releasing soon.  There are a few reasons we keep our own step counter

1. A run can have multiple tfevent files with different global_steps
2. If a user sets sync_tensorboard=True and also wants to call wandb.log to send custom artifacts this makes it simpler.
3. Tensorboard steps can go backwards, currently our step counter must increase

Our integration adds a “global_step” key for each tensorboard namespace.  In the UI you can choose any step you’ve logged as an x-axis for your charts.

We’re working on an overhaul of our metrics backend to address these issues.

Here’s my branch fixing the log error / better handling multiple tfevent namespaces: https://github.com/wandb/client/pull/596/files

**Comment by vanpelt** [comment]
We made some quick fixes in 0.8.14 and this message will no longer be displayed.
